src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/ControlPlaneDataStore.cs

```diff
@@ -66,14 +67,24 @@ public async Task<IdentityProvider> GetIdentityProviderAsync(string name, Cancel
             return identityProvider.ToIdentityProvider();
         }
 
-        public async Task<IdentityProvider> UpsertIdentityProviderAsync(IdentityProvider identityProvider, CancellationToken cancellationToken)
+        public async Task<IEnumerable<IdentityProvider>> GetAllIdentityProvidersAsync(CancellationToken cancellationToken)
+        {
+            return await GetSystemDocumentsByIdAsync<CosmosIdentityProvider>(null, CosmosIdentityProvider.IdentityProviderPartition, cancellationToken);
```

Comment:
Do the response from this look like the cosmos versions of the documents or the ones that get sent in? I've had trouble in the past with the implicit conversion which is why the `ToIdentityProvider()` extension exists.